src/Core/Validation/FieldValidation/NumericFieldValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Core\Validation\FieldValidation\FieldValidator;
+
+class NumericFieldValidator extends FieldValidator
+{
+    /**
+     * Minimum size of the number
+     */
+    private ?int $minValue;
+
+    /**
+     * Maximum size of the number
+     */
+    private ?int $maxValue;
+
+    public function __construct(
+        string $name,
+        mixed $value,
+        ?int $minValue = null,
+        ?int $maxValue = null
+    ) {
+        $this->minValue = $minValue;
+        $this->maxValue = $maxValue;
+        parent::__construct($name, $value);
+    }
+
+    protected function validateValue(): ValidationResult
+    {
+        $result = ValidationResult::create();
+        if (!is_numeric($this->value) || is_string($this->value)) {
+            // Must be a numeric value, though not as a numeric string
+            $message = _t(__CLASS__ . '.WRONGTYPE', 'Must be numeric');
+            $result->addFieldError($this->name, $message, value: $this->value);
+            return $result;
+        } elseif (isset($this->minValue) && $this->value < $this->minValue) {
```

Comment:
```suggestion
        } elseif ($this->minValue !== null && $this->value < $this->minValue) {
```
We really only need to check for null.

`isset()` does a bit more than that and it seems like there's more going on. Semantically `isset()` is "have we set a value for that?". In this case, we _have_ set a value for that, it's just that the value we've set may be null.

Please update the one below as well.

Reply:
Updated to use is_null() which is functionally the same and my preferred way to do this comparison